### PR TITLE
Work : 1359928 Add Express Route Connector json and examples.  [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by dedhar**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (olydis) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Properties `location`, `tags` and `name` are inherited by `Resource`! Don't specify those here.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (fearthecowboy) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: You've duplicated 'tags' from Resource

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Moving to using subresource. Is there any advantage or disadvantage in using resource vs subresource

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: I modelled this new json on RouteFilter.json which uses subresource

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: I am moving to using SubResource in allOf. That should also resolve the build errors

**Comment by dedhar** [comment]
Resolved olydis and fearthecowboy comments to move to using Resource and remove redundant fields : tags etc. Vishrut mentioned the build failure is due to Mode being Ruby. He kicked off another build

**Comment by dedhar** [comment]
I have linked my Microsoft account to azure GitHub. How do I get rid of cla required label? @olydis / @fearthecowboy : any suggestions on how to resolve this?

**Comment by dedhar** [comment]
Also in the first phase of this project we are planning to make authorizations optional parameter. If all circuits are in same subscription we don't need authorization.

**Comment by fearthecowboy** [comment]
Adding @dsgouda  as reviewer (this is reopened from #1430 on wrong branch)

**Comment by dedhar** [comment]
@fearthecowboy  : From the old PR : 

we've reorganized this repository a bit -- can I get you to make a clean fork of the repository, and base your PR off the current branch instead of `master, and then issue a new PR?

So I got a clean fork from current :
D:\One\Networking\azure-rest-api-specs>git remote -v
Azure   https://github.com/Azure/azure-rest-api-specs.git (fetch)
Azure   https://github.com/Azure/azure-rest-api-specs.git (push)
origin  https://github.com/dedhar/azure-rest-api-specs.git (fetch)
origin  https://github.com/dedhar/azure-rest-api-specs.git (push)

D:\One\Networking\azure-rest-api-specs>git branch
* current

Do I need new PR?

**Comment by fearthecowboy** [comment]
@dedhar  I think this PR is OK -- it's running thru the CI checks and it's based against `current` and the files are in the right spots.

**Comment by dedhar** [comment]
Thanks @fearthecowboy !  Let me know if there are additional comments.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteCircuit.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteCircuit.json]: nit: formatting

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteCircuit.json]: addressed in latest commit

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: you don't need to explicitly define this

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: you mean x-ms-long-running-operation? Will address

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: 4XX status codes are not supported by autorest, please use default instead. You can do this however 
https://github.com/Azure/azure-rest-api-specs/blob/2fb9a0b3b902335ff0b0033711c234431931ec9d/specification/sql/resource-manager/Microsoft.Sql/2015-05-01-preview/virtualNetworkRules.json#L55

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: same comment as before for 4xx status codes

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: If you already have a separate _Update operation please make this _Create

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: same comment as above for 4xx codes.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Please mark this as readonly

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: please provide ```x-ms-examples``` for this operation

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed in latest commit

**Comment by dedhar** [comment]
Addressed following comments in latest commit : 
- Example for list by subscription
- Remove 4xx error codes

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed in latest commit

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed in latest commit

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed

**Comment by dedhar** [comment]
Addressed latest comments from Deepak

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: still needs fixing

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: sorry about that

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed and checked examples as well

**Comment by dedhar** [comment]
addressed latest comments from Deepak

**Review by dsgouda — COMMENTED** [review]
Looks good for the most part, per @dedhar this needs ARM approval. @ravbhatnagar ?

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Please model the default error response.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Error response missing.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: This looks like it contains secret information. It should not be returned in the response of GET/PUT. Expose this using the POST api.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (ravbhatnagar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Looks like the intended use of this property is different from provisioningState. Please make this an enum. Also add "provisioningState" property

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: SDK Team requested me to remove 4xx Error codes.
@fearthecowboy / @olydis / @dsgouda  : Can you comment if we need to add back error codes?

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: SDK Team requested me to remove 4xx Error codes.
@fearthecowboy / @olydis / @dsgouda  : Can you comment if we need to add back error codes?

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: The customer specifies authorization Key as part of PUT API Call and we just intend to return whatever customer specified in ExpressRouteConnector object. Authorization Key is also returned as part of VirtualNetworkGatewayConnectionObject : Reference : virtualNetworkGateway.json

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: I checked with team internally here and for now we just need provisioningState. When we expand the API to accept list of circuits we would consider exposing connectionStatus for each link between a pair of circuits

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: The error response is denoted by ```"default"```. please [refer](https://github.com/Azure/azure-rest-api-specs/blob/2fb9a0b3b902335ff0b0033711c234431931ec9d/specification/apimanagement/resource-manager/Microsoft.ApiManagement/2016-10-10/apimapis.json#L81)
@ravbhatnagar please confirm

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Renamed connectionStatus to provisioningState in latest commit

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed in latest commit

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: removed authorization keys from GET / LIST calls

**Comment by dedhar** [comment]
Addressed Gaurav's comments to remove authorization keys from GET / LIST calls and added 4xx error codes to default error response

**Comment by dedhar** [comment]
Addressed following comments from Gaurav : Remove authorizations from PUT / PATCH Call and default error responses

**Comment by dedhar** [comment]
@ravbhatnagar / @dsgouda  : Can you take a look? Once I have approval, I would then move the code to https://github.com/Azure/azure-rest-api-specs-pr

**Comment by dedhar** [comment]
Sample ARM Deployment Template : 
ExpressRouteConnector.json

{
    "$schema": "https://schema.management.azure.com/schemas/2017-08-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "expressRouteConnectorName": {
            "type": "string",
            "metadata": {
                "description": "This is the name of the Express Route Connector Resource"
            }
        },
        "location": {
            "type": "string",
            "defaultValue": "West US",
            "metadata": {
                "description": "Azure Region where Express Route Connector resource be deployed."
            }
        },
        “circuits”: {
            “type”:”array”,
“defaultValue”:[       "/subscriptions/CustomerSubscriptionUS/resourceGroups/ExpressRouteResourceGroupUS/providers/Microsoft.Network/expressRouteCircuits/ExpressRouteARMCircuitA",
	"/subscriptions/CustomerSubscriptionAus/resourceGroups/ExpressRouteResourceGroupAus/providers/Microsoft.Network/expressRouteCircuits/ExpressRouteARMCircuitB"
              ],
	“metadata”:”List of Express Route Circuits with complete URI. Not needed if all circuits are in same subscription.”
         },
        “authorizations”: {
“type”:”array”,
“defaultValue”:[     
     "abcdefgh^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
	      "^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^"
              ],
	“metadata”:”List of Authorization Keys on all express route circuits being peered”
        },
        “addressPrefix”: {
               “type”:”string”
                “defaultValue”:”10.0.0.0/24”.
                 “metadata”:”/24 subnet to be used for iBGP Peering.”
         }
    },
    "resources": [
      {
        "apiVersion": "2017-08-01",
        "location": "[parameters('location')]",
        "name": "[parameters('filterName')]",
        "properties": {
        },
        "tags": {
          "key1": "value1",
          "key2": "value2"
        },
        "type": "Microsoft.Network/expressRouteConnectors"
      }
    ]
}

**Comment by salameer** [comment]
Hi @ravbhatnagar 

Please review from ARMs side

**Comment by dedhar** [comment]
I see that Tracis CI build passed. @dsgouda / @ravbhatnagar   : Were there more comments to address? Not seeing any new comments. :)

**Comment by ravbhatnagar** [comment]
There is just one outstanding issue around returning secrets in the response of PUT/PATCH that is currently being discussed.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Please add description here

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Review by dsgouda — COMMENTED** [review]
The 2 new errors introduced are due to models inheriting from ```Resource``` model defined in network.json which is incorrectly modeled.
The warnings can be ignored as discussed offline. 
Pending the minor requested change, LGTM subject to ARM approval.

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: addressed in latest commit

**Comment by dedhar** [comment]
addressed latest comment from Deepak

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: Did you mean ```description```?

Comment (dedhar) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json]: oh yes since you had recommended to add description in previous comment

**Review by dsgouda — APPROVED** [review]
LGTM pending ARM approval.

**Comment by ravbhatnagar** [comment]
#ARM sign off

**Comment by salameer** [comment]
What's the deal here folks, is this the one where a Private PR was created instead?

**Comment by dsgouda** [comment]
@salameer yes there is a PR on the private repo which is under review.

**Comment by dsgouda** [comment]
@dedhar please close this PR if you plan to implement the service and open a new PR in the future.

**Comment by salameer** [comment]
@dsgouda what's the status here?

**Comment by dsgouda** [comment]
PR on the private repo is merged, I believe @dedhar and her team is now working on implementing the APIs, per my last comment, I am awaiting to know if they'd like to create a new public PR in the future.

**Comment by salameer** [comment]
Hi  @dsgouda, I guess my question here is, is there any reason for this to stay open?

**Comment by dsgouda** [comment]
Not at my end no.

**Pull request closed without merging**
